packages/block-library/src/tag-cloud/edit.js

```diff
@@ -0,0 +1,141 @@
+	PanelBody,
+	Placeholder,
+	ToggleControl,
+	SelectControl,
+	ServerSideRender,
+	Toolbar,
+} from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { BlockControls, InspectorControls } from '@wordpress/editor';
+
+class TagCloudEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			editing: ! this.props.attributes.taxonomy,
+		};
+
+		this.setTaxonomy = this.setTaxonomy.bind( this );
+		this.toggleShowTagCounts = this.toggleShowTagCounts.bind( this );
+	}
+
+	getTaxonomyOptions() {
+		const taxonomies = filter( this.props.taxonomies, 'show_cloud' );
+		const selectOption = {
+			label: __( '- Select -' ),
+			value: '',
+		};
+		const taxonomyOptions = map( taxonomies, ( taxonomy ) => {
+			return {
+				value: taxonomy.slug,
+				label: taxonomy.name,
+			};
+		} );
+
+		return [ selectOption, ...taxonomyOptions ];
+	}
+
+	setTaxonomy( taxonomy ) {
+		const { setAttributes } = this.props;
+
+		this.setState( { editing: false } );
+
+		setAttributes( { taxonomy } );
+	}
+
+	toggleShowTagCounts() {
+		const { attributes, setAttributes } = this.props;
+		const { showTagCounts } = attributes;
+
+		setAttributes( { showTagCounts: ! showTagCounts } );
+	}
+
+	render() {
+		const { editing } = this.state;
+		const { attributes } = this.props;
+		const { taxonomy, showTagCounts } = attributes;
+		const taxonomyOptions = this.getTaxonomyOptions();
+
+		const toolbarControls = [
+			{
+				icon: 'edit',
+				title: __( 'Edit Taxonomy' ),
+				onClick: () => this.setState( { editing: true } ),
+			},
+		];
+
+		const blockControls = (
+			<BlockControls>
+				<Toolbar controls={ toolbarControls } />
+			</BlockControls>
+		);
+
+		const inspectorControls = (
+			<InspectorControls>
+				<PanelBody title={ __( 'Tag Cloud Settings' ) }>
+					<SelectControl
+						label={ __( 'Taxonomy' ) }
+						options={ taxonomyOptions }
+						value={ taxonomy }
+						onChange={ this.setTaxonomy }
+					/>
+					<ToggleControl
+						label={ __( 'Show post counts' ) }
+						checked={ showTagCounts }
+						onChange={ this.toggleShowTagCounts }
+					/>
+				</PanelBody>
+			</InspectorControls>
+		);
+
+		if ( ! taxonomy || editing ) {
+			return (
+				<Fragment>
+					{ inspectorControls }
+					<Placeholder
+						key="placeholder"
+						icon="tag"
+						label={ __( 'Tag Cloud' ) }
+					>
+						<SelectControl
+							label={ __( 'Select a Taxonomy' ) }
+							value={ taxonomy }
+							options={ taxonomyOptions }
+							onChange={ this.setTaxonomy }
+						/>
+					</Placeholder>
+				</Fragment>
+			);
+		}
+
+		return (
+			<Fragment>
+				{ blockControls }
+				{ inspectorControls }
+				<ServerSideRender
+					key="tag-cloud"
+					block="core/tag-cloud"
+					attributes={ attributes }
+				/>
+			</Fragment>
+		);
+	}
+}
+
+export default withSelect( ( select ) => {
+	return {
+		taxonomies: select( 'core' ).getTaxonomies( { per_page: -1 } ),
```

Comment:
Is `per_page` a valid argument for fetching taxonomies, and if so, is `-1` a valid value?

Reply:
Yes, this query param will be ignored by the REST API. 

```suggestion
		taxonomies: select( 'core' ).getTaxonomies(),
```